python/ql/src/semmle/python/RegexTreeView.qll

```diff
@@ -836,6 +836,13 @@ class RegExpZeroWidthMatch extends RegExpGroup {
  */
 class RegExpSubPattern extends RegExpZeroWidthMatch {
   RegExpSubPattern() { not re.emptyGroup(start, end) }
+
+  /** Gets the lookahead term. */
+  RegExpTerm getOperand() {
+    result.getRegex() = re and
+    result.getStart() = start + 3 and
+    result.getEnd() = end - 1
+  }
```

Comment:
This is slightly incorrect. A `RegExpZeroWidthMatch` can arise from `re.negativeAssertionGroup(start, end)`, where `end in start+[3,4]`.

I think you want something like:
```suggestion
  /** Gets the lookahead term. */
  RegExpTerm getOperand() {
    exists(int in_start, int in_end | re.groupContents(start, end, in_start, in_end) |
      result.getRegex() = re and
      result.getStart() = in_start and
      result.getEnd() = in_end
    )
  }
```